### DISCUSSION TITLE
Using 2D boolean functions on drawings

### DISCUSSION THREAD

**Discussion by raydeleu** [sgenoud/replicad#110] [Show and tell]
For the Replicad manual at https://github.com/raydeleu/ReplicadManual/tree/main#readme I prepared a small example how the 2D boolean functions can be used. In the code sample below I combined this with selecting edges on the two extremities of one of the shapes, using a combination of `inPlane` finders. 

I noticed that `intersect` also works. I still have to find an example where this function would be useful.  

![image](https://github.com/sgenoud/replicad/assets/38007983/2a63392a-64db-4d4e-b168-02efcd37dd61)

``` javascript
const { draw, drawCircle, drawRectangle} = replicad;

const main = () => {
let axleRadius = 11
let keySlotHeight = 6
let keySlotWidth  = 2.50  

let axleHole = drawCircle(axleRadius)
let axleHole2 = drawCircle(axleRadius).translate(3*axleRadius,0)
let keySlot  = drawRectangle(2*keySlotWidth,keySlotHeight)
.translate(-axleRadius,0)
let keySlot2  = drawRectangle(2*keySlotWidth,keySlotHeight)
.translate(-axleRadius,0).translate(3*axleRadius,0)
let axleShape = axleHole.cut(keySlot).sketchOnPlane("XZ")
let axleShape2 = axleHole2.fuse(keySlot2).sketchOnPlane("XZ",20)
let axle = axleShape.extrude(25)
let axle2 = axleShape2.extrude(25)
            .fillet(2,(e)=>e.either([
                          (e)=>e.inPlane("XZ",45),
                          (e)=>e.inPlane("XZ",20)
])
)
  return [axle,axle2];
};
```

**Comment by raydeleu** [comment]
A useful application of the `.intersect()` method is creating a curved slot. The following image illustrates the chosen approach. 

<img width="1254" alt="sector_intersection1" src="https://github.com/sgenoud/replicad/assets/38007983/76958583-adaa-496e-9736-fd094d7314fa">

**Reply by sgenoud** [reply]
this does look great!

**Comment by sgenoud** [comment]
Your reference doc is great! The more I think about it, the more it should live in your own space! Would you like some help to automate some kind of build to have a page? Would you be interested in a replicad hostname to point to it (something like manual.replicad.xyz?

**Reply by raydeleu** [reply]
Thanks for the compliment. I do not know what you mean exactly with an automated build of a page. If you refer to the system that you are using to create you documentation I would hesitate as it looks complicated to maintain. 

I am quite happy with the github support of uploading images and displaying an asciidoc file. I prefer to use asciidoc at it can be converted to other formats as well. Being old-fashioned I like the option to create a large pdf with the complete manual, having chapter numbers, having references to illustrations and chapters, having an automated table of contents etc. Markdown comes close, but is mostly aimed at making websites, not documents. The only downside of using github is that the document may not include other files, so it has to live in one big readme file. 

Of course I would welcome if a replicad hostname points to the manual, as I hope that the manual is not only useful to me but also to others. Perhaps we can attract more users to Replicad in this way. I think compared to tools like FreeCAD it could offer a more straightforward access to advanced CAD tools for the "maker-community".

**Reply by sgenoud** [reply]
>  If you refer to the system that you are using to create you documentation I would hesitate as it looks complicated to maintain.

I was thinking about something different that would build a simple website (+pdf perhaps) directly from the asciidoc and run from your github repo.

**Reply by raydeleu** [reply]
That would be perfect!
